application/src/main/kotlin/net/corda/v5/application/flows/FlowContextProperties.kt

```diff
@@ -0,0 +1,46 @@
+package net.corda.v5.application.flows
+
+/**
+ * Context properties of a [Flow] are key value pairs of Strings. They are comprised of two sets of properties, those
+ * set by the platform when a [Flow] is started, and those which are added by the CorDapp developer during the execution
+ * of their flow. The latter set are referred to as user context properties. All context properties are immutable once
+ * set.
+ * Where keys exist as platform properties already, setting user properties will always be ignored, only the platform
```

Comment:
Is there a reason why we ignore setting user properties if a platform one exists? Would it be better to throw an exception so its clear what the behaviour is?

Reply:
I did think about this and I can't come up with a good answer, except this avoids runtime exceptions. It's not possible to guess whether that's better than some unusual behaviour.

Something else we could suggest here is that users always scope their properties like packages. We could then do the same, e.g. `net.corda.v5.account`?

I'm open to suggestions.

Reply:
@JamesHR3 @mnesbit Any thoughts?

Reply:
Decided offline to throw. Also throw on any "corda." properties.